vllm/v1/kv_cache_interface.py

```diff
@@ -206,9 +208,14 @@ def page_size_bytes(self) -> int:
         return page_size
 
     def max_memory_usage_bytes(self, vllm_config: VllmConfig) -> int:
-        # We allocate 1 block for each request now, so max_memory_usage_bytes is
-        # the same as page_size_bytes.
-        # Need to update this when supporting prefix caching.
+        if self.cache_strategy == "last":
+            # Keeps the last full block and one non-full block state:
+            return 2 * self.page_size_bytes
+        elif self.cache_strategy == "all":
+            # Keeps a state at every block boundary:
+            max_model_len = vllm_config.model_config.max_model_len
```

Comment:
you can free the initial blocks so this is too over-estimated. I think it should be similar to SlidingWindowSpec.max_memory_usage_bytes